mdagent/tools/base_tools/thermo_tools.py

```diff
@@ -0,0 +1,198 @@
+    name = "ComputeIsothermalCompressabilityKappaT"
+    description = """Compute the isothermal compressibility (kappa_T) for
+      a molecular dynamics trajectory. Requires a trajectory file and
+      temperature and optionally a topology file. Returns the isothermal
+      compressibility."""
+    path_registry: PathRegistry | None = None
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _run(self, traj_file: str, temperature: float, top_file: str | None = None):
+        # todo -> might need to convert inputs to the right types
+        traj = load_traj(self.path_registry, traj_file, top_file)
+        if not traj:
+            return "Failed to load trajectory file."
+        return str(md.isothermal_compressability_kappa_T(traj, temperature))
+
+    async def _arun(
+        self, traj_file: str, temperature: float, top_file: str | None = None
+    ):
+        raise NotImplementedError("Async version not implemented")
+
+
+class ComputeThermalExpansionAlphaP(BaseTool):
+    name = "ComputeThermalExpansionAlphaP"
+    description = """Compute the thermal expansion coefficient (alpha_P)
+    for a molecular dynamics trajectory. Requires a trajectory file and
+      temperature and energies as an array containing the potential
+      energies of each trajectory frame, in units of kJ/mol. Optionally
+      requires a topology file. Returns the thermal expanssion
+      coefficient, units of inverse Kelvin."""
+    path_registry: PathRegistry | None = None
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _run(self, traj_file, top_file=None, temperature=300, energies=None):
+        # todo -> might need to convert inputs to the right types
+        traj = load_traj(self.path_registry, traj_file, top_file)
+        if not traj:
+            return "Failed to load trajectory file."
+        if not energies:
+            return (
+                "Energies must be provided to compute the thermal"
+                "expansion coefficient."
+            )
+        if energies.shape[0] != traj.n_frames:
+            return (
+                "The length of the energies array must match the "
+                "number of frames in the trajectory."
+            )
+        return str(md.thermal_expansion_alpha_P(traj, temperature, energies))
+
+    async def _arun(self, traj_file, top_file=None, temperature=300, energies=None):
+        raise NotImplementedError("Async version not implemented")
+
+
+class ComputeDensity(BaseTool):
+    name = "ComputeDensity"
+    description = """Calculate the mass density of each frame in a
+    trajectory. Requires a trajectory file and an array of masses and
+    optionally a topology file. Returns an array of mass densities for
+    each frame."""
+    # todo -> should this write to a file or return the array?
```

Comment:
since its each frame

- plot it over the frames
- write it to a file (source data for the plot is accessible
- return to the agent some message about it